File: emulinker/src/main/java/org/emulinker/kaillera/model/event/UserQuitGameEvent.java
package org.emulinker.kaillera.model.event;

import org.emulinker.kaillera.model.*;

public class UserQuitGameEvent implements GameEvent {
  private KailleraGame game;
  private KailleraUser user;

  public UserQuitGameEvent(KailleraGame game, KailleraUser user) {
    this.game = game;
    this.user = user;
  }

  @Override
  public String toString() {
    return "UserQuitGameEvent";
  }

  @Override
  public KailleraGame getGame() {
    return game;
  }

  public KailleraUser getUser() {
    return user;
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/model/event/UserQuitGameEvent.kt
package org.emulinker.kaillera.model.event

import org.emulinker.kaillera.model.KailleraGame
import org.emulinker.kaillera.model.KailleraUser

// If game is null, I think that means the user quit the whole server.
class UserQuitGameEvent(override val game: KailleraGame?, val user: KailleraUser) : GameEvent {
  override fun toString(): String {
    return "UserQuitGameEvent"
  }
}
